tests/js/server/aql/aql-functions-list.js

```diff
@@ -847,21 +847,41 @@ function ahuacatlListTestSuite () {
 ////////////////////////////////////////////////////////////////////////////////
 
     testRemoveNthInvalid : function () {
-      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH()"); 
-      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH([ ])"); 
-      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH([ ], 1, true)"); 
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH()");
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH([ ])");
+      assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN REMOVE_NTH([ ], 1, true)");
 
       assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN NOOPT(REMOVE_NTH())");
       assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN NOOPT(REMOVE_NTH([ ]))");
       assertQueryError(errors.ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH.code, "RETURN NOOPT(REMOVE_NTH([ ], 1, true))");
-    }
+    },
+
+
+    testInterleave: function () {
+      const data = [
+        {input: [[], []], expected: []},
+        {input: [[1, 1], [2, 2]], expected: [1, 2, 1, 2]},
+        {input: [[1, 3, 5], [2, 4, 6]], expected: [1, 2, 3, 4, 5, 6]},
+        {input: [[1, 3, 5], [2]], expected: [1, 2, 3, 5]},
+        {input: [[1, 1, 1], [2, 2], [3]], expected: [1, 2, 3, 1, 2, 1]},
+        {input: [[1, 4, 7], [2, 5], [3]], expected: [1, 2, 3, 4, 5, 7]},
+        {input: [[1],[1],[1],[1],[1],[1],[1]], expected: [1, 1, 1, 1, 1, 1, 1]},
```

Comment:
Can you also add a test where you have many arrays, but only one is empty?
The first test can not be distinguished from we return a empty array if we see any empty array in input.